app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/pocket/PocketStoriesComposables.kt

```diff
@@ -153,20 +153,25 @@ fun PocketStories(
  * Displays a list of [PocketRecommendedStoriesCategory]s.
  *
  * @param categories The categories needed to be displayed.
+ * @param selections List of categories currently selected.
  * @param onCategoryClick Callback for when the user taps a category.
+ * @param modifier [Modifier] to be applied to the layout.
  */
 @Composable
 fun PocketStoriesCategories(
     categories: List<PocketRecommendedStoriesCategory>,
     selections: List<PocketRecommendedStoriesSelectedCategory>,
-    onCategoryClick: (PocketRecommendedStoriesCategory) -> Unit
+    onCategoryClick: (PocketRecommendedStoriesCategory) -> Unit,
+    modifier: Modifier = Modifier
 ) {
-    StaggeredHorizontalGrid(
-        horizontalItemsSpacing = 16.dp
-    ) {
-        categories.filter { it.name != POCKET_STORIES_DEFAULT_CATEGORY_NAME }.forEach { category ->
-            SelectableChip(category.name, selections.map { it.name }.contains(category.name)) {
-                onCategoryClick(category)
+    Box(modifier = modifier) {
```

Comment:
This makes me wonder when we should pass `Modifier` as a param. Personally, I would opt to inline the modifier in this instance until `PocketStoriesCategories` needs to be used in multiple places where its layout needs to be modified  for each of the different views. Otherwise, I find it easier to see what the layout parameters are if it was inlined. Passing modifiers also becomes a bit trickier if for example we needed another modifier param for `StaggeredHorizontalGrid`. Usually I am thinking about how we can keep the list of params optimal given its usage.

I had a similar debate with doing this when I was adding `RecentTabImage` https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt#230 where it did make sense that we want to pass in different sizes for the image.

Reply:
> This makes me wonder when we should pass `Modifier` as a param. Personally, I would opt to inline the modifier in this instance until `PocketStoriesCategories` needs to be used in multiple places where its layout needs to be modified for each of the different views. Otherwise, I find it easier to see what the layout parameters are if it was inlined. Passing modifiers also becomes a bit trickier if for example we needed another modifier param for `StaggeredHorizontalGrid`. Usually I am thinking about how we can keep the list of params optimal given its usage.
> 
> I had a similar debate with doing this when I was adding `RecentTabImage` https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt#230 where it did make sense that we want to pass in different sizes for the image.

I don't think we need to necessary change this. I think this is more food for thought and can help us align with what we feel is a good practise.

Reply:
Thank you!
Indeed I also though about this a little.
What I went with is allowing the ViewHolder to inject the padding to each composable the same as is being done in XML (where we have a top level paddings / margins set)
For Pocket we needed to go one level lower - set the spacing to each layout child instead of the layout parent because we needed to set this on the LazyRow. But for the other two composables ViewHolders I set the spacing in the ComposeView - the same as we'd do in XML for the parent layout.
The idea being that the ViewHolder controls the dimensions of each composable. And then each composable is free to layout how it wants in those dimensions.